### PR TITLE
Hotfix 1.27.16

### PR DESCRIPTION
**Pull request by arteymix**
A few things to check:

 - [x] test that "disease model" effectively appear as a choice for term category
 - [x] release baseCode 1.1.3 to get the null-handling naive ebayes implementation

A few things to cherry-pick:

 - [x] fix security contexts for all runnables and callables 466f2c472cc4981a4e5ea4f4c228504d18e60151
 - [x] fix missing security context in CLI batch operations a4d43374c4aa4adc36c5c7d6a50a4816f17ea77 df7c4932bf878c7cf7462aacaf177956f7effff16
 - [x] fix removal of gene products e233a98a00365ee3aa2dc9b695f9e490920d963c
 - [x] revert the task executor change for NCBI gene loader/converter 1fa29219efc13d637d17262b7ba55a12583bd854 and f9bdf5b12b7d5c1fa30d439cb35059f90843a6b5
 - [x] finish merging 1.27.15 in the dev trunk https://github.com/PavlidisLab/Gemma/pull/373
 - [x] https://github.com/PavlidisLab/Gemma/pull/381
 - [x] 50b2d9c75917af13d104f2beb2433ee2792fa638
 - [x] 852fe2a4557a6442686b40fa7f4cda2ae9106062
 - [ ] db295684f75a0ac403a28fb258067c64881e8fb2
 - [x] d328a1df01850dc35cc8f30b9f11a627bc9bac0f
 - [x] 0ef00461b9975504c46198804edf60aa42300bd1

### CODE REVIEW

File: gemma-core/src/main/resources/ubic/gemma/core/ontology/EFO.factor.categories.txt [hunk 1]
```diff
@@ -13,6 +13,7 @@ http://www.ebi.ac.uk/efo/EFO_0000410	disease staging
 http://www.ebi.ac.uk/efo/EFO_0000434	ecotype
 http://www.ebi.ac.uk/efo/EFO_0000470	environmental stress
 # no longer allowed:
+http://gemma.msl.ubc.ca/ont/TGEMO_00101 disease model
```
Comment (arteymix) [inline gemma-core/src/main/resources/ubic/gemma/core/ontology/EFO.factor.categories.txt:16]: This still needs to be tested manually.

File: gemma-core/src/test/java/ubic/gemma/core/search/SearchServiceTest.java [hunk 2]
```diff
@@ -143,7 +144,7 @@ public void tearDown() {
      * are found, -- requires LARQ index.
      */
     @Test
-    public void testGeneralSearch4Brain() throws SearchException {
+    public void testGeneralSearch4Brain() throws SearchException, OntologySearchException {
```
Comment (arteymix) [inline gemma-core/src/test/java/ubic/gemma/core/search/SearchServiceTest.java:147]: This should not be necessary, the search service does not throw OntologySearchException.

File: gemma-core/src/main/java/ubic/gemma/core/apps/NCBIGene2GOAssociationLoaderCLI.java [hunk 3]
```diff
@@ -66,7 +65,7 @@ protected void buildOptions( Options options ) {
     @Override
     protected void doWork() throws Exception {
         TaxonService taxonService = this.getBean( TaxonService.class );
-        AsyncTaskExecutor taskExecutor = this.getBean( AsyncTaskExecutor.class );
+        TaskExecutor taskExecutor = this.getBean( TaskExecutor.class );
```
Comment (arteymix) [inline gemma-core/src/main/java/ubic/gemma/core/apps/NCBIGene2GOAssociationLoaderCLI.java:68]: To be reverted.

File: gemma-core/src/main/java/ubic/gemma/core/apps/NcbiGeneLoaderCLI.java [hunk 4]
```diff
@@ -100,7 +99,7 @@ protected boolean requireLogin() {
 
     @Override
     protected void doWork() throws Exception {
-        loader = new NcbiGeneLoader( this.getBean( AsyncTaskExecutor.class ) );
+        loader = new NcbiGeneLoader( this.getBean( TaskExecutor.class ) );
```
Comment (arteymix) [inline gemma-core/src/main/java/ubic/gemma/core/apps/NcbiGeneLoaderCLI.java:102]: To be reverted.

**Comment by arteymix** [comment]
@ppavlidis is d328a1df01850dc35cc8f30b9f11a627bc9bac0f something we want to include in here?

**Comment by ppavlidis** [comment]
Yes as ncbi has sequence version numbers like 2.1 now but it can wait if need be, it's vital for gene info update but as that was done recently it's not urgent

**Comment by arteymix** [comment]
Hmm, sorry for the confusion, I meant db295684f75a0ac403a28fb258067c64881e8fb2.

**Comment by arteymix** [comment]
Ok, it's all done but https://github.com/PavlidisLab/Gemma/commit/db295684f75a0ac403a28fb258067c64881e8fb2. I'm testing that disease model are effective appearing in the UI an then merging this into master.

File: gemma-core/src/test/java/ubic/gemma/core/search/SearchServiceTest.java (hunk 2 above)
Comment (arteymix) [reply gemma-core/src/test/java/ubic/gemma/core/search/SearchServiceTest.java:147]: It's actually used with OntologyService, everything is ok!

**Pull request merged**